vcd/resource_vcd_vapp.go

```diff
@@ -494,6 +495,16 @@ func resourceVcdVAppDelete(d *schema.ResourceData, meta interface{}) error {
 
 	_ = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
 		task, err := vapp.Undeploy()
+		// Very often the vApp is powered off at this point and Undeploy() waits as long as
+		// vcdClient.MaxRetryTimeout before just going to Delete phase. vCD documentation makes it hard to comprehend
+		// when an Undeploy() is actually needed, but the below error string is a 100% match:
+		// "The requested operation could not be executed since vApp __vApp_name__ is not running"
+		// If the error matches we just quit retrying and fast forward to vapp.Delete()
```

Comment:
Suggesting to rephrase to:

// Very often the vApp is powered off at this point and Undeploy() would wait for as long as
// vcdClient.MaxRetryTimeout before going to Delete phase because of the following error:
// "The requested operation could not be executed since vApp __vApp_name__ is not running"
// So, if the error matches we just quit retrying and fast forward to vapp.Delete()